python/scripts/load_data.py

```diff
@@ -20,208 +20,403 @@
+        if meta_path is None:
+            meta_path = os.path.abspath(os.path.join(_scripts_path,
+                                                     'meta.json'))
+        if manifest_path is None:
+            manifest_path = os.path.abspath(os.path.join(_scripts_path,
+                                                         'manifest.csv'))
+        self._keep_temp_files = keep_temp_files
+
+        # load given meta.json and manifest.csv files into memory
+        self.meta = ingestionfunctions.load_meta_data(meta_path)
+        self.manifest = ManifestReader(manifest_path)
+
+        # setup engine for database_choice
+        self.engine = dbtools.get_database_engine(self.database_choice)
+
+        # update the meta.json to the database
+        ingestionfunctions.meta_json_to_database(self.engine, self.meta)
+
+    def drop_tables(self):
+        """
+        Returns the outcome of dropping all the tables from the 
+        database_choice and then rebuilding.
+        """
+        logging.info("Dropping all tables from the database!")
+        db_conn = self.engine.connect()
+        query_result = list()
+        query_result.append(db_conn.execute(
+            "DROP SCHEMA public CASCADE;CREATE SCHEMA public;"))
+
+        if self.database_choice == 'remote_database' or self.database_choice \
+                == 'remote_database_master':
+            query_result.append(db_conn.execute('''
+            GRANT ALL PRIVILEGES ON SCHEMA public TO housingcrud;
+            GRANT ALL PRIVILEGES ON ALL TABLES    IN SCHEMA public TO housingcrud;
+            GRANT ALL PRIVILEGES ON ALL SEQUENCES IN SCHEMA public TO housingcrud;
+            GRANT ALL ON SCHEMA public TO public;
+            '''))
+        return query_result
+
+    def update_only(self, these_data):
+        """
+        Reloads only the flat file associated to the unique_data_id in the
+        list 'these_data'.
+        """
+        logging.info("update_only(): attempting to update {} data".format(
+            these_data))
+        processed_data_id = []
+
+        for manifest_row in self.manifest:
+            data_id = manifest_row['unique_data_id']
+            use = manifest_row['include_flag'] == 'use'
+
+            # process manifest row for requested data_id if flagged for use
+            if use and data_id in these_data:
+                logging.info("Manifest row found for {} - preparing to "
+                             "load data.".format(data_id))
+
+                temp_filepath = self._get_temp_filepath(
+                    manifest_row=manifest_row)
+
+                # get objects for interfacing with the database
+                sql_interface = self._configure_db_interface(
+                    manifest_row=manifest_row, temp_filepath=temp_filepath)
+                sql_manifest_row = sql_interface.get_sql_manifest_row()
+
+                # if data_id has status = loaded in sql_manifest
+                try:
```

Comment:
I think that to combine load_all_data with update_only, we'll want to put all the stuff in this 'try' block into a `_remove_existing_data` method and wrap that in an if statement to only run if the user says to update the data. 

In addition - the best way to handle this is probably file by file. If we add a column to the manifest ('delete_existing') then that can be the flag as to whether to delete or not.